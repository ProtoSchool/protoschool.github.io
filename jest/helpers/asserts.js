// ** SNAPSHOT FILTERING ***
// prevent snapshots for logs that change on each instance, such as shuffled quiz options

const excludedPhrases = [
  "Cool! Here's what we get when we mix up the order of the answer choices:"
]

// returns true if log includes a string from the excludedPhrases array
function isExcluded (log) {
  return excludedPhrases.some(phrase => log.includes(phrase))
}

// add log to snapshot UNLESS it contains an excluded phrase
export function assertLogSnapshot (log) {
  if (!isExcluded(log)) {
    expect(log).toMatchSnapshot() // expect if log isn't excluded from assertion
  }
  // do nothing if log is excluded from assertion
}
