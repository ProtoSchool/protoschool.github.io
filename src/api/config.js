/* eslint-disable no-unused-expressions */
export const staticPath = process.env.DATA_STATIC_PATH || 'src/static'
export const tutorialsPath = process.env.DATA_TUTORIALS_PATH || 'src/tutorials'

const boilerplatesPath = process.env.LESSON_BOILERPLATES_PATH || `${tutorialsPath}/boilerplates`

export const boilerplates = {
  path: boilerplatesPath,
  markdownPath: `${boilerplatesPath}/boilerplate.md`,
  challengeMarkdownPath: `${boilerplatesPath}/boilerplate-challenge.md`,
  multipleChoiceJsPath: `${boilerplatesPath}/boilerplate-multiple-choice.js`,
  codeJsPath: `${boilerplatesPath}/boilerplate-code.js`,
  fileUploadJsPath: `${boilerplatesPath}/boilerplate-file-upload.js`
}
