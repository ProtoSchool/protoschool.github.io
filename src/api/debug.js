export default (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test') && process.env.DEBUG
