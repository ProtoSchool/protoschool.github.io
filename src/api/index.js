/**
 * API for ProtoSchool.
 *
 * TODO: Make API universal https://github.com/ProtoSchool/protoschool.github.io/issues/589
 *  - make all scripts use this API
 *  - change it to be universal so the website can also use it
 *   - ATM we have deps on fs.readFile etc
 *
 * @module api
 */
module.exports = {
  projects: require('./modules/projects'),
  lessons: require('./modules/lessons'),
  courses: require('./modules/courses').default,
  resources: require('./modules/resources'),
  tutorials: require('./modules/tutorials'),
  _config: require('./config')
}
