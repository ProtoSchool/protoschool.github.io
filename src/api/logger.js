export const debug = (...args) => {
  console.log('⚙️ api', ...args)
}

export const createLogGroup = module => functionMethod => `[${module}.${functionMethod}()]`
