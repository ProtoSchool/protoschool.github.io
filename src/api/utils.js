import { writeFileSync } from 'fs'
import { resolve } from 'path'
import { staticPath } from './config.js'

const correctedCases = {
  api: 'API',
  cid: 'CID',
  of: 'of',
  a: 'a'
}

export function writeStaticFile(staticFileName, data) {
  writeFileSync(resolve(staticPath, staticFileName), JSON.stringify(data, null, 2))
}

export function deriveShortname(path) {
  return path.split('-').map(word => (
    correctedCases[word] ? correctedCases[word] : (word.charAt(0).toUpperCase() + word.slice(1))
  )).join(' ')
}
