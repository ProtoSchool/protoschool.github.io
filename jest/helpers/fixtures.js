import { lessons as _lessons, resources as _resources, courses, tutorials } from '../../src/api'

// Creators: create data (tutorials, lessons)
export function createTutorial (config = { override: {}, lessons: 0, resources: 0 }) {
  const { tutorial, lessons, resources } = generateTutorial(config)
  const createdTutorial = tutorials.create(tutorial)

  for (let i = 0; i < lessons.length; ++i) {
    _lessons.create(tutorials.get(createdTutorial.id), lessons[i])
  }

  resources.forEach(resource => {
    _resources.add(createdTutorial.id, resource)
  })

  return tutorials.get(createdTutorial.id)
}

// Generators: generate data to be used

export function generateTutorial (config = { override: {}, lessons: 0, resources: 0, lessonOverride: {} }) {
  const suffix = tutorials.getNextTutorialId()

  const tutorial = {
    title: `New Tutorial (${suffix})`,
    url: `new-tut-${suffix}`,
    project: 'libp2p',
    description: 'New tutorial description',
    ...config.override
  }

  const lessons = new Array(config.lessons).fill().map((_, i) => (
    generateLesson({
      override: {
        title: `Lesson ${i + 1}`,
        ...config.lessonOverride
      }
    }).lesson
  ))

  const resources = new Array(config.resources).fill().map((_, i) => (
    generateResource({
      override: {
        title: `Resource ${i + 1}`,
        link: `https://resource${i + 1}.com`
      }
    }).resource
  ))

  return {
    tutorial,
    lessons,
    resources,
    expected: { ...tutorial, lessons, resources }
  }
}

export function generateLesson ({ createTutorial = false, override = {} } = {}) {
  const lesson = {
    title: 'Lesson',
    type: 'text',
    ...override
  }
  let tutorial

  if (createTutorial) {
    tutorial = tutorials.create(generateTutorial().tutorial)
    courses.add(tutorial.id)
  }

  return {
    lesson,
    tutorial,
    expected: { lesson, tutorial }
  }
}

export function generateResource ({ createTutorial, override = {} } = {}) {
  const resource = {
    title: 'Resource',
    link: 'https://resource.com',
    description: 'Resoure description',
    type: 'demo',
    ...override
  }
  let tutorial

  if (createTutorial) {
    tutorial = tutorials.create(generateTutorial().tutorial)
    courses.add(tutorial.id)
  }

  return {
    resource,
    tutorial,
    expected: { resource, tutorial }
  }
}
