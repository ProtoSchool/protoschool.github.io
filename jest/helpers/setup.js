import { courses, tutorials } from '../../src/api'

export function restoreData (lastTutorialId) {
  const newLastTutorialId = tutorials.list.getLatest().id

  // delete all new tutorials
  for (let id = lastTutorialId + 1; id <= newLastTutorialId; id++) {
    tutorials.remove(id)
    courses.remove(id)
  }
}
