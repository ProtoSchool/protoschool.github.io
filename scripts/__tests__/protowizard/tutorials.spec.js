import { tutorials } from '../../../src/api'
import { restoreData } from '../../../jest/helpers/setup'
import { generateTutorial } from '../../../jest/helpers/fixtures'
import { assertNewTutorial } from '../helpers/asserts'
import { protowizard } from '../helpers/runners'

describe('protowizard', () => {
  let lastTutorialId

  beforeAll(() => {
    lastTutorialId = tutorials.list.getLatest().id
  })

  afterEach(() => {
    restoreData(lastTutorialId)
  })

  describe('1. create tutorial', () => {
    test('1.1. should create tutorial (skips lesson creation)', async () => {
      const { tutorial, expected } = generateTutorial()

      await protowizard([
        { type: 'tutorial' },
        tutorial,
        { confirm: false } // no to create first lesson
      ])

      assertNewTutorial({
        context: { lastTutorialId },
        expected,
        result: tutorials.getByUrl(tutorial.url)
      })
    })

    test('1.2. should create tutorial with one text lesson', async () => {
      const { tutorial, lessons, expected } = generateTutorial({
        lessons: 1
      })

      await protowizard([
        { type: 'tutorial' },
        tutorial,
        { confirm: true }, // yes to create the first lesson
        lessons[0],
        { confirm: false }, // no to create another lesson
        { confirm: false } // no to create the first resource
      ])

      assertNewTutorial({
        context: { lastTutorialId },
        expected,
        result: tutorials.getByUrl(tutorial.url)
      })
    })

    test('1.3. should create tutorial with two text lessons', async () => {
      const { tutorial, lessons, expected } = generateTutorial({
        lessons: 2
      })

      await protowizard([
        { type: 'tutorial' },
        tutorial,
        { confirm: true }, // yes to create the first lesson
        lessons[0],
        { confirm: true }, // yes to create another lesson
        lessons[1],
        { confirm: false }, // no to create another lesson
        { confirm: false } // no to create the first resource
      ])

      assertNewTutorial({
        context: { lastTutorialId },
        expected,
        result: tutorials.getByUrl(tutorial.url)
      })
    })

    test('1.4. should create tutorial with one lesson and one resource', async () => {
      const { tutorial, lessons, resources, expected } = generateTutorial({
        lessons: 1,
        resources: 1
      })

      await protowizard([
        { type: 'tutorial' },
        tutorial,
        { confirm: true }, // yes to create the first lesson
        lessons[0],
        { confirm: false }, // no to create another lesson
        { confirm: true }, // yes to create the first resource
        resources[0],
        { confirm: false } // no to create another resource
      ])

      assertNewTutorial({
        context: { lastTutorialId },
        expected,
        result: tutorials.getByUrl(tutorial.url)
      })
    })

    test('1.5. should create tutorial with one lesson and two resources', async () => {
      const { tutorial, lessons, resources, expected } = generateTutorial({
        lessons: 1,
        resources: 2
      })

      await protowizard([
        { type: 'tutorial' },
        tutorial,
        { confirm: true }, // yes to create the first lesson
        lessons[0],
        { confirm: false }, // no to create another lesson
        { confirm: true }, // yes to create the first resource
        resources[0],
        { confirm: true }, // yes to create another resource
        resources[1],
        { confirm: false } // no to create another resource
      ])

      assertNewTutorial({
        context: { lastTutorialId },
        expected,
        result: tutorials.getByUrl(tutorial.url)
      })
    })

    test('1.6. should not allow to create two tutorials with the same title or url', async () => {
      const { tutorial } = generateTutorial()

      await protowizard([
        { type: 'tutorial' },
        tutorial,
        { confirm: false } // no to create first lesson
      ])

      await expect(
        protowizard([
          { type: 'tutorial' },
          tutorial,
          { confirm: false } // no to create first lesson
        ])
      ).rejects.toThrow('INQUIRER VALIDATION FAILED')
    })
  })
})
