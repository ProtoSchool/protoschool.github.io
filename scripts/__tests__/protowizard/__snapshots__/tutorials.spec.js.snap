// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protowizard 1. create tutorial 1.1. should create tutorial (skips lesson creation) 1`] = `"Let's create the files you need to build your tutorial. I'll ask you a few questions to get started."`;

exports[`protowizard 1. create tutorial 1.1. should create tutorial (skips lesson creation) 2`] = `"Thanks! I've created a directory for your tutorial at \`src/tutorials/0011-new-tut-11/\`."`;

exports[`protowizard 1. create tutorial 1.1. should create tutorial (skips lesson creation) 3`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.1. should create tutorial (skips lesson creation) 4`] = `"Okay, no problem. You can summon me later to add lessons."`;

exports[`protowizard 1. create tutorial 1.1. should create tutorial (skips lesson creation) 5`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 2`] = `"Let's create the files you need to build your tutorial. I'll ask you a few questions to get started."`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 3`] = `"Thanks! I've created a directory for your tutorial at \`src/tutorials/0011-new-tut-11/\`."`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 4`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 5`] = `
"Tada! I've created the following files that you'll need for this lesson:
 ‣ 01.md (for writing the text of your lesson)"
`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 6`] = `"To preview this lesson, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 7`] = `
"Okay, sounds like we're done. Here are all the lessons now included in \\"New Tutorial (11)\\":
 ‣ 1 - Lesson 1 (text)"
`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 8`] = `"You can find all the files you'll need for these lessons in the \`src/tutorials/0011-new-tut-11/\` directory."`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 9`] = `"All tutorials have a resources page where users can find opportunities for further learning."`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 10`] = `"Okay, no problem. You can summon me later to add resources."`;

exports[`protowizard 1. create tutorial 1.2. should create tutorial with one text lesson 11`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 2`] = `"Let's create the files you need to build your tutorial. I'll ask you a few questions to get started."`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 3`] = `"Thanks! I've created a directory for your tutorial at \`src/tutorials/0011-new-tut-11/\`."`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 4`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 5`] = `
"Tada! I've created the following files that you'll need for this lesson:
 ‣ 01.md (for writing the text of your lesson)"
`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 6`] = `"To preview this lesson, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 7`] = `
"Tada! I've created the following files that you'll need for this lesson:
 ‣ 02.md (for writing the text of your lesson)"
`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 8`] = `"To preview this lesson, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/02"`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 9`] = `
"Okay, sounds like we're done. Here are all the lessons now included in \\"New Tutorial (11)\\":
 ‣ 1 - Lesson 1 (text)
 ‣ 2 - Lesson 2 (text)"
`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 10`] = `"You can find all the files you'll need for these lessons in the \`src/tutorials/0011-new-tut-11/\` directory."`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 11`] = `"All tutorials have a resources page where users can find opportunities for further learning."`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 12`] = `"Okay, no problem. You can summon me later to add resources."`;

exports[`protowizard 1. create tutorial 1.3. should create tutorial with two text lessons 13`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 2`] = `"Let's create the files you need to build your tutorial. I'll ask you a few questions to get started."`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 3`] = `"Thanks! I've created a directory for your tutorial at \`src/tutorials/0011-new-tut-11/\`."`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 4`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 5`] = `
"Tada! I've created the following files that you'll need for this lesson:
 ‣ 01.md (for writing the text of your lesson)"
`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 6`] = `"To preview this lesson, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 7`] = `
"Okay, sounds like we're done. Here are all the lessons now included in \\"New Tutorial (11)\\":
 ‣ 1 - Lesson 1 (text)"
`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 8`] = `"You can find all the files you'll need for these lessons in the \`src/tutorials/0011-new-tut-11/\` directory."`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 9`] = `"All tutorials have a resources page where users can find opportunities for further learning."`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 10`] = `"I've added \\"Resource 1\\" to your resources list."`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 11`] = `
"Okay, sounds like we're done. Here are all the resources now included in \\"New Tutorial (11)\\":
 ‣ Resource 1 (demo)"
`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 12`] = `"To preview your resources page, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/resources"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 13`] = `"Need to change something? You can edit your resources in the file \`src/static/tutorials.json\`."`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 14`] = `"Awesome work! \\"New Tutorial (11)\\" has both lesson files and resources!"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 15`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 16`] = `"To create the content of your lessons, edit the files in the \`src/tutorials/0011-new-tut-11/\` directory. (Learn more at https://bit.ly/protoschool-content.)"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 17`] = `"To update your tutorial's title, description, or resources, edit its entry in the \`src/static/tutorials.json\` file. (Learn more at https://bit.ly/protoschool-metadata.)"`;

exports[`protowizard 1. create tutorial 1.4. should create tutorial with one lesson and one resource 18`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 2`] = `"Let's create the files you need to build your tutorial. I'll ask you a few questions to get started."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 3`] = `"Thanks! I've created a directory for your tutorial at \`src/tutorials/0011-new-tut-11/\`."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 4`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 5`] = `
"Tada! I've created the following files that you'll need for this lesson:
 ‣ 01.md (for writing the text of your lesson)"
`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 6`] = `"To preview this lesson, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/01"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 7`] = `
"Okay, sounds like we're done. Here are all the lessons now included in \\"New Tutorial (11)\\":
 ‣ 1 - Lesson 1 (text)"
`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 8`] = `"You can find all the files you'll need for these lessons in the \`src/tutorials/0011-new-tut-11/\` directory."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 9`] = `"All tutorials have a resources page where users can find opportunities for further learning."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 10`] = `"I've added \\"Resource 1\\" to your resources list."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 11`] = `"I've added \\"Resource 2\\" to your resources list."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 12`] = `
"Okay, sounds like we're done. Here are all the resources now included in \\"New Tutorial (11)\\":
 ‣ Resource 1 (demo)
 ‣ Resource 2 (demo)"
`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 13`] = `"To preview your resources page, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/resources"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 14`] = `"Need to change something? You can edit your resources in the file \`src/static/tutorials.json\`."`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 15`] = `"Awesome work! \\"New Tutorial (11)\\" has both lesson files and resources!"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 16`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 17`] = `"To create the content of your lessons, edit the files in the \`src/tutorials/0011-new-tut-11/\` directory. (Learn more at https://bit.ly/protoschool-content.)"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 18`] = `"To update your tutorial's title, description, or resources, edit its entry in the \`src/static/tutorials.json\` file. (Learn more at https://bit.ly/protoschool-metadata.)"`;

exports[`protowizard 1. create tutorial 1.5. should create tutorial with one lesson and two resources 19`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 1`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 2`] = `"Let's create the files you need to build your tutorial. I'll ask you a few questions to get started."`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 3`] = `"Thanks! I've created a directory for your tutorial at \`src/tutorials/0011-new-tut-11/\`."`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 4`] = `"To preview your tutorial, first run \`npm start\` in a separate terminal window or tab, then visit this page in your web browser: http://localhost:3000/new-tut-11/"`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 5`] = `"Okay, no problem. You can summon me later to add lessons."`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 6`] = `"View the detailed guide to developing tutorials at: https://bit.ly/protoschool-developing"`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 7`] = `"Welcome! I'm the ProtoWizard, and I'm excited to help you build your ProtoSchool tutorial."`;

exports[`protowizard 1. create tutorial 1.6. should not allow to create two tutorials with the same title or url 8`] = `"Let's create the files you need to build your tutorial. I'll ask you a few questions to get started."`;
